### DISCUSSION TITLE
Theme

### DISCUSSION THREAD

**Discussion by uma-uc** [chakra-ui/zag#111] [General]
This is an interesting approach indeed.
How do u support switching multiple themes using css attribute selector

**Comment by anubra266** [comment]
Hi @uma-uc, this is something that'll be handled by the design systems built on zag. Zag aims to handle the component primitives.

**Reply by uma-uc** [reply]
What I meant was, as an example in solid-ts, accordion component,
injectGlobal(accordionStyle) is injected prior. So if the theme changes how should the changes be applied?

**Reply by anubra266** [reply]
@uma-uc, those styles are just injected for the purpose of our demos, you can use any design systems of your choice that could help you handle color modes.

**Reply by uma-uc** [reply]
I see - thanks for your prompt response. Would u be able to provide an example?  Are you saying we pass as class and not use css attribute selector?

**Reply by anubra266** [reply]
@uma-uc take the component sections like normal HTML tags, zag is just providing the html attributes to handle their logic and accessibility, you can now style them yourself by passing HTML classes. 
What framework are you better familiar with? Maybe that would help me explain better.

**Reply by uma-uc** [reply]
got it.. Thanks @anubra266  - I saw the example in zag-docs/components/machines/accordion.tsx as example... using hope-ui

**Comment by segunadebayo** [comment]
Closing as outdated — Zag is a headless library and does not include theming.
